server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -3924,10 +3924,7 @@ protected void commitIndexWriter(final IndexWriter writer, final Translog transl
             Loggers.addAppender(LogManager.getLogger(Engine.class), mockLogAppender);
 
             // Issue the first flushOnIdle request. The flush happens in the background using the flush threadpool.
```

Comment:
nit: the comment should be moved together with the `shard.flushOnIdle(0)` call?

Reply:
Well, the problem was that the addExpectation was done after the flushOnIdle() call which made the test fail. It needs to be before. So I made a comment about the whole "paragraph" of code, which combines the explanation of doing a flushOnIdle but also adding the expectation for the log message. So moving the comment afterwards would not explain the log expectation, that is why I prefer it is there above the "paragraph".